src/storage/settings/settingsActions.ts

```diff
@@ -47,3 +51,37 @@ export const walletLockTimeChanged = createAction<Settings['general']['walletLoc
 export const analyticsToggled = createAction<Settings['general']['analytics']>('settings/analyticsToggled')
 
 export const fiatCurrencyChanged = createAction<Settings['general']['fiatCurrency']>('settings/fiatCurrencyChanged')
+
+export const receiveTestnetTokens = createAsyncThunk<PendingTransaction, AddressHash, { rejectValue: SnackbarMessage }>(
+  'assets/receiveTestnetTokens',
+  async (destinationAddress: AddressHash, { rejectWithValue, fulfillWithValue }) => {
+    const response = await fetch('https://faucet.testnet.alephium.org/send', {
+      method: 'POST',
+      body: destinationAddress
+    })
+
+    if (!response.ok) {
+      return rejectWithValue({
+        text:
+          response.status === 429
+            ? i18n.t('You have reached the maximum calls limit. Please try again in a few minutes.')
+            : i18n.t('Encountered error while calling the faucet.'),
+        type: 'alert'
+      })
+    }
+
+    const responseURL = await (await response.text()).trim()
+
+    const hash = responseURL.match(/\/([a-fA-F0-9]+)$/)?.[1] || ''
+
+    return fulfillWithValue({
+      hash: hash,
+      fromAddress: 'Faucet',
+      toAddress: destinationAddress,
+      amount: '12',
+      timestamp: new Date().getTime(),
+      status: 'pending',
+      type: 'transfer'
+    })
```

Comment:
GJ! 👏 

nit: How about assigning the return value to a variable called `pendingTransaction` first, just for readability's sake?